MFVNeutralino/test/predict2v_newsimpl.py

```diff
@@ -0,0 +1,175 @@
+from JMTucker.Tools.ROOTTools import *
+from JMTucker.Tools.general import *
+import pandas as pd
+
+presel_path = '/uscms_data/d2/tucker/crab_dirs/PreselHistosV27m'
+#sel_path = '/uscms_data/d3/dquach/crab3dirs/HistosV27m_moresidebands'
+sel_path = '/uscms/home/pkotamni/nobackup/crabdirs/Histos2trkmergedV27m'
+data = bool_from_argv('data')
+year = '2017' if len(sys.argv) < 2 else sys.argv[1]
+varname = 'nom' if len(sys.argv) < 3 else sys.argv[2] # use the BTV variations to compute syst shifts on pred2v
+print("variation: %s" % varname)
+
+# turn this on to get the proper pred2v stat err
+# FIXME eventually just use this in place of epred
+print_pred_n2v_propagated_stat_err = True
+
+if data:
+    fn, presel_scale = 'JetHT%s.root' % year, 1.
+else:
+    fn, presel_scale = 'background_%s.root' % year, 1.
+
+def propagate_product(x, y, ex, ey):
+    p = x * y
+    e = p * ((ex / x)**2 + (ey / y)**2)**0.5
+    return e
+
+def fb(ft,efft,frt):
+    return (ft-frt)/(efft-frt)
+
+presel_f = ROOT.TFile(os.path.join(presel_path, fn))
+sel_f = ROOT.TFile(os.path.join(sel_path, fn))
+effs = pd.read_csv('MiniTree/efficiencies/all_effs.csv',index_col='variant')
+cb_vals = pd.read_csv('One2Two/cb_vals/cb_vals.csv',index_col='variant')
+
+fracdict = {}
+cdict = {}
+presel_var = 'presel_%s_%s' % (year, varname)
+fracdict['presel'] = fb(effs.at[presel_var,'ft'], effs.at[presel_var,'efft'], effs.at[presel_var,'frt'])
+
+for ntk in 3,4,5,7,8,9:
+    fracdict[ntk] = {}
+    cdict[ntk] = {}
+
+    var = '%strk_1v_%s_%s' % (ntk, year, varname)
+    try:
+        fracdict[ntk]['1v'] = fb(effs.at[var,'ft'], effs.at[var,'efft'], effs.at[var,'frt'])
+    except:
+        print "Warning: did not find effs for",ntk
+        fracdict[ntk]['1v'] = 0
+
+    year_formatted = year if not data else "data_%s" % year
+    cb_label = '%s_%strk' % (year_formatted, ntk)
+    try:
+        cdict[ntk]['cb'] = cb_vals.at[cb_label+'_cb','cb_val']
+        cdict[ntk]['cbbar'] = cb_vals.at[cb_label+'_cbbar','cb_val']
+    except:
+        print "Warning: did not find cb/cbbar for",ntk
+        cdict[ntk]['cb'] = 0
+        cdict[ntk]['cbbar'] = 0
+
+npresel, enpresel = get_integral(presel_f.Get('mfvEventHistosJetPreSel/h_npu'))
+npresel  *= presel_scale
+enpresel *= presel_scale
+f0 = fracdict['presel']
+
+print 'year:', year
+print 'presel events: %8.0f +- %4.0f' % (npresel, enpresel)
+print 'fraction of presel events with b-quarks: %.3f' % f0
+print '%16s %8s %8s %8s %19s %15s %35s' % ('n1v', 'f1', 'cb', 'cbbar', 'pred n2v', 'n2v', 'ratio')
+
+sum_n1v = 0
+alpha_nn = {}
+sum_rest = 0
+sum2_erest = 0
+sum_n2v = 0
+sum2_en2v =0
+sum2_en1v = 0
```

Comment:
It might be useful to specify (in a comment?) what these sums are / why we use them. Some of them are not obvious to me from a quick glance, and I'm sure it'll be even less obvious to either of us in a few months from now!

Reply:
Okay, I will add comments on these new variables and point to the evernote doc.